content/concepts/payment-system-basics/accounts/cryptographic-keys.md

```diff
@@ -116,6 +116,106 @@ The supported types of key pairs can be used interchangeably throughout the XRP
+
+    **Tip:** Validators use this root key pair. If you are calculating a validator's key pair, you can stop here. To distinguish between these two different types of public keys, the [base58][] serialization for validator public keys uses the prefix `0x1c`.
+
+2. Convert the root public key to its 33-byte compressed form.
+
+    The uncompressed form of any ECDSA public key consists of a pair of 32-byte integers: an X coordinate, and a Y coordinate. The compressed form is just the X coordinate and a one-byte prefix: `0x02` if the Y coordinate is even, or `0x03` if the Y coordinate is odd.
+
+    You can convert an uncompressed public key to the compressed form with the `openssl` commandline tool. For example, if the uncompressed public key is in the file `ec-pub.pem`, you can output the compressed form like this:
+
+        $ openssl ec -in ec-pub.pem -pubin -text -noout -conv_form compressed
+
+3. Derive an "intermediate key pair" from the compressed root public key you, as follows:
+
+    1. Concatenate the following in order, for a total of 40 bytes:
+        - The compressed root public key (33 bytes)
+        - `0x00000000000000000000000000000000` (4 bytes of zeroes). (This value was intended to be used to derive different members of the same family, but in practice only the value 0 is used.)
+        - A "key sequence" value (4 bytes), as a big-endian unsigned integer. Use 0 as a starting value for the key sequence.
+
+    2. Calculate the [SHA-512Half][] of the concatenated value.
+
+    3. If the result is not a valid secp265k1 private key, increment the key sequence by 1 and restart deriving the account's intermediate key pair.
+
+    4. With a valid secp256k1 private key, use the standard ECDSA public key derivation with the secp256k1 curve to derive the intermediate public key. (As always with cryptographic algorithms, use a standard, well-known, publicly-audited implementation whenever possible. For example, [OpenSSL](https://www.openssl.org/) has implementations of core Ed25519 and secp256k1 functions.)
+
+4. Derive the master public key pair by adding the intermediate public key to the root public key. Similarly, derive the private key by adding the intermediate private key to the root private key.
+
+    - An ECDSA private key is just a very large integer, so you can calculate the sum of two private keys by summing them modulo the secp256k1 modulus.
+
+    - An ECDSA public key is a point on the elliptic curve, so you should use elliptic curve math to sum the points.
+
+    **Tip:** You don't need any private keys to derive the master public key. You can do so using only the root public key.
```

Comment:
I'd probably leave this tip off. That's a useful fact when we're deriving a family of keys, and we have a machine that needs to calculate public keys from the family, but we don't want to put secret keys on that machine. However, we don't use that functionality at all.